### PR TITLE
For #18507 - Ensure credit card screens cannot be screenshot

### PR DESCRIPTION
**Pull request by chrishernan**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorFragment.kt [hunk 1]
```diff
@@ -50,6 +47,13 @@ class CreditCardEditorFragment : Fragment(R.layout.fragment_credit_card_editor)
         )
     }
 
+    override fun onResume() {
+        requireActivity().window.setFlags(
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorFragment.kt:51]: nit: we seem to use `addFlags`/`clearFlags` in other places, might be best to follow the same pattern there too.

**Comment by gabrielluong** [comment]
@Mugurell can you take a look at this? I am not too familiar with this area. Thanks!

File: app/src/main/java/org/mozilla/fenix/settings/creditcards/CreditCardsSettingFragment.kt [hunk 2]
```diff
@@ -65,4 +77,19 @@ class CreditCardsSettingFragment : PreferenceFragmentCompat() {
...
+        removeFlagSecure()
+        super.onStop()
+    }
+
+    /**
+     * Checks var isNextDestinationCreditCardEditor to decide whether to remove or keep
+     * WindowManager.LayoutParams.FLAG_SECURE
+     */
+    private fun removeFlagSecure() {
+        if (!isNextDestinationCreditCardEditor) {
```
Comment (Mugurell) [inline app/src/main/java/org/mozilla/fenix/settings/creditcards/CreditCardsSettingFragment.kt:91]: Seeing that `CreditCardEditorFragment` sets `FLAG_SECURE` on it's own (which is probably desired) maybe we don't need this if check and have `onStop` always remove the flag?

File: app/src/main/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorFragment.kt [hunk 3]
```diff
@@ -50,6 +51,13 @@ class CreditCardEditorFragment : Fragment(R.layout.fragment_credit_card_editor)
         )
     }
 
+    override fun onResume() {
+        activity?.window?.setFlags(
```
Comment (Mugurell) [inline app/src/main/java/org/mozilla/fenix/settings/creditcards/CreditCardEditorFragment.kt:55]: As on `CreditCardSettingFragment` setting `FLAG_SECURE` here might be too late.
Ideally we should set this even before of `super.onStart` and then remove that flag in `onStop`.

**Review by Mugurell — CHANGES_REQUESTED** [review]
Thank you for helping with this.
It generally looks good but given the importance of this feature maybe the code can be a little tighter. 
Also, to prevent against future regressions a few simple unit ensuring `FLAG_SECURE` exists at the expected lifecycle callbacks tests would be nice. We can help with this if needed.

**Comment by gabrielluong** [comment]
Gonna open this up to allow the team to address the feedback.

**Pull request closed without merging**